### ISSUE TITLE
4.7.6. - 3/4ths of all plugins are missing... Scanner crashes...

### ISSUE THREAD

**Issue by Lykdraft** [kmatheussen/radium#955]
Hey there,

nice work on the PManager. However 3/4rths of all my plugins are now missing in the VST/AU lists.
When I use the new PManager to scan them, Radium scans a loooong time, then crashes.

Two problems here: 

1. when I restart Radium, the scan starts from scratch, not where I left off/Radium crashed. 
Usually the last plugin that crashes a DAW gets "quarantined"/"blacklisted" or "ignored" on rescan
and the scan goes on from there.

2. You don't see which plugin is currently scanned or get a notification which plugin crashes Radium.
When having plenty of plugins there is **no** way to know which plugin causes Radium to crash.

Basically forces me to stay put on 4.7.4 before this is solved... ;)

Best,
tL.

**Comment by Lykdraft** [comment]
_(By the way: The PM should be able to accept "Enter" to add a plugin, or a keycommand for the "add" button. You can search by keyboard, tab to the results, use cursors to go up/down in the list but then need to click "add". Feels strange to me.)_

**Comment by kmatheussen** [comment]
But you can't do anything more in 4.7.4 than you could in 4.7.6. Radium has never scanned all plugins before, or had an option to scan all plugins. Radium has never stored the result of scanning plugins either. It's always been scan-by-need only.

**Comment by kmatheussen** [comment]
Just to be clear. I'm not saying that it will always be like this, I'm just saying that 4.7.6 is not a regression.

**Comment by Lykdraft** [comment]
Okay. Seems legit. But I have no access to the lists anymore.They are broken now. 

I **have** to use the PM now, which is useful as a search "engine" but not as a browser, right?
So why are the lists now broken? (When everything is like before, just the PManager on top of it...) 

_(Otherwise you might want to ditch the lists completely and go PManager only, because nearly empty lists don't make much sense, right? :) )_

Also: is there an option to get the PM popped up via a ra.command?

**Comment by kmatheussen** [comment]
Well, the PM should show the exact same content as the plugin popup menus?
What do you mean when you say "list"? (i.e in "lists now broken"?)

You can start the plugin manager like this:
```
F12 CTRL_L : ra.evalScheme "(pmg-start (make-instrument-conf) (lambda x x))"
```
(but this might not work in later versions)

**Comment by Lykdraft** [comment]
> F12 CTRL_L : ra.evalScheme "(pmg-start (make-instrument-conf) (lambda x x))"

Ah good. 

> Well, the PM should show the exact same content as the plugin popup menus?
What do you mean when you say "list"? (i.e in "lists now broken"?)

I mean the Plugin Popup Menu Lists. (i.e. VST, VST3, AU) that were used before to select plugins.
These show only a fraction of all my plugins from before. (i.e. in every version **before 4.7.4)

I thought that this has something to do with the PluginManager, as I have lost acess to the majority of all my plugins **except** through the new plugin manager...

Hope that makes sense.

**Comment by kmatheussen** [comment]
Oh, I don't know why that has happened. But that's a separate bug, not related to the plugin manager.
The plugin menu code was completely rewritten in 4.7.6 though, in order to make it async.

**Comment by Lykdraft** [comment]
Oh my... I suppose I need more sleep... I know what's going on. You have limited the numbers of shown plugins. There is now this little "next" item at the very end of each page. Lol...

_I suppose this is on purpose? (Instead of showing all plugs over the whole screen like before?)_

So that's why I thought they are missing there. All good.

**Comment by kmatheussen** [comment]
Don't think I meant to change anything actually. Are you talking about the favorite plugins in the bottom of the main menu, or are you talking about the VST/VST3/AU sub menues? And, is it better or worse now? :-)

**Comment by Lykdraft** [comment]
No, I mean the "usual" VST/VST3/AU Menus. Actually, it is both. Worse and better... :)  
It is worse because when you access plugins through the VST submenu you have to go a couple of times "next" to reach plugins which letters are higher up in the alphabet or lowecase. (lowercase is at the end)

It is better, because AU plugins don't sort by vendor/creator, so they are flat and thus some are offscreen.

Bottomline: I think it is worse because a "birds eye overview" of all plugins was pretty nice before and
b) the plugin manager can be used to get to specific plugins that are AU only and harder to get to.

As I like speed of access more that anything else I would say, yeah, no 6x browsing "next" to
get to the end of the list... ;)

**Comment by Lykdraft** [comment]
_(The "next" option for the favorites is fine because otherwise, I **think** the right-click menu would get very cluttered up...)_

**Comment by kmatheussen** [comment]
Is it "Next" or "next" by the way? If it says "next" (lowercase 'n'), then something very strange is going on, because "next" (lowercase 'n') is only used for the favourite plugins.

But all popup menus in the program have always been configured so that if there is 200 entries, we put the rest into "Next". Is this number lower than 200 now? Or have the 200 limit actually started to work?

**Comment by Lykdraft** [comment]
its "Next".  Seems like the limit kicked in indeed. And it does look like a 150-170 limit.
With big screens and over 1K plugins that might be a bit low though... 
(I have 7x next to go to the end here..lol)

_Any chance this can be a configurable option?_

![blank_skitch_document](https://cloud.githubusercontent.com/assets/18437561/25547041/c49c7158-2c65-11e7-8bdf-18297fc4f642.jpg)

**Comment by Lykdraft** [comment]
> _Any chance this can be a configurable option?_

With that I mean the 200 limit...

**Comment by kmatheussen** [comment]
Everything is clear to me now. The old code used some custom code to create the plugin menu, while the new code uses the default code. Yes, it's possible to make the limit a configurable option.

**Comment by Lykdraft** [comment]
Awesome. Shall I close this then and make a separate thread with reference to this one?

**Comment by kmatheussen** [comment]
It's quite funny though. I added the 200 limit specifically for plugins filling up the screen, but to the wrong code. Now, it's finally working. :-)

Yes, pleas make a separate thread.

**Comment by Lykdraft** [comment]
Haha, okay alright... :)
